sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -76,6 +79,50 @@
 DEFAULT_BUNDLE_PROCESSOR_CACHE_SHUTDOWN_THRESHOLD_S = 60
 
 
+class ShortIdCache(object):
+  """ Cache for MonitoringInfo "short ids"
+  """
+  def __init__(self):
+    self._lock = threading.Lock()
+    self._lastShortId = 0
+    self._infoKeyToShortId = {}  # type: Dict[FrozenSet, str]
+    self._shortIdToInfo = {}  # type: Dict[str, metrics_pb2.MonitoringInfo]
+
+  def getShortId(self, monitoring_info):
+    """ Returns the assigned shortId for a given MonitoringInfo, assigns one if
+    not assigned already.
+    """
+    # type: (metrics_pb2.MonitoringInfo) -> str
+    with self._lock:
+      key = monitoring_infos.to_key(monitoring_info)
```

Comment:
I think you missed adding your monitoring_infos changes.

Reply:
This function exists already: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/metrics/monitoring_infos.py#L329-L338

Reply:
yup